### DISCUSSION TITLE
Restart/reconnect to bluetooth amplifier

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#1683] [Bug Reports]
@rern 

RPI: 3B+
Base: i20240408
Update: 20240603

For sometime I have been using a bluetooth (Anker SoundCore2) speaker with one of my RPi (3A) systems with great success. I can use the speaker buttons to power the speaker off and the 3A detects this and shuts down rAudio.  When I power the speaker up again, rAudio also detects this and restarts the BT connection and starts the audio output - great (I have 'Features / AutoPlay / Bluetooth connected' enabled).

I have recently purchased a Fosi Audi BT10A bluetooth amp to use with another RPI3B+ based system.  This works really well, so long as the amplifier remains on.  I have enabled the same Features / Autoplay as above. If I turn the amp off, rAudio detects the BT source has disconnected, turns off the audio output and disconnects the BT connection - so far. so good  When I power up the amp again, rAudio in this situation, does not reconnect the BT source and so no audio output is heard.

I can force it to connect using:

bluetoothctl connect <BT Mac address>
mpc play

rAudio will then play as normal.

Can the Bluetooth Features / Autostart processes be amended to reconnect to the Fosi ?

Please let me know if you need any further information from me.

Thanks

Chris

**Comment by ceejayemm** [comment]
@rern 
Since the update to 20240612, the Anker SoundCore 2 BT Speaker which was connecting and playing correctly when the BT speaker is turned on - now DOES NOT.  

With the web app started I can see the SoundCore 2 being connected, the Network page shows it connected via Bluetooth but it does not start to play automatically as it used to.  On the Player page I can see that the Music Player Daemon is starting but is not now being set to play.  The Features / Other / AutoPlay settings are still set to Bluetooth Connected (Enabled) and Power On / Reboot (Enabled)..

The system still refuses to connect/play when the Fosi amp is started in Bluetooth mode, as originally reported.

Updating to 20240615 makes no difference either.

Chris

**Reply by rern** [reply]
Please try the UPDATE.

**Reply by ceejayemm** [reply]
@rern 
I have been away from home for a couple of days so I have only just been able to try this for you.

The latest UPDATE branch seems to bring the RPi 3A / SoundCode2 setup back to where it used to be. Turning the SoundCore 2 on starts the 3A rAudio installation playing as I would expect.

Unfortunately the same isn't true for the RPi 3B+ / Fosi amp combination. The Fosi has been previously manually paired with rAudio. Turning on the Fosi amp, the BT announcement (if its there) is not detected by rAudio and so no audio output is heard.  If I use the Connect function from rAudio Networks / Bluetooth then a connection is made and I get sound output.  I am beginning to think that the Fosi is not sending the necessary BT command/announcements when it is powered so rAudio doesn't know its there.  Is there anyway to confirm this, ie some sort of command that would show any/all BT commands being sent from the Fosi and received by rAudio even if that are not the right ones ?

Regards

Chris

**Comment by ceejayemm** [comment]
@rern 
I see that you have made some further changes in 20240706/07 to the Bluetooth Autoplay feature.  Unfortunately this doesn't seem to make any difference to this problem at the moment.

Regards.

Chris
